### PR TITLE
Ability to specify extra headers for the request

### PR DESCRIPTION
**Pull request by andrewebdev**
This adds the functionality for the issue outlined on #34

### CODE REVIEW

**Comment by andrewebdev** [comment]
I signed it!

File: iron-form.html [hunk 1]
```diff
@@ -76,6 +76,16 @@
       withCredentials: {
         type: Boolean,
         value: false
+      },
+
+      /**
+      * Specify extra headers to be sent with the ajax request
```
Comment (notwaldorf) [inline iron-form.html]: Looking at the property in `iron-ajax`, it also has the comment "Note: setting a `Content-Type` header here will override the value specified by the `contentType` property of this element.", which I think we should also add here.

**Comment by notwaldorf** [comment]
@andrewebdev hiya! friendly ping to see if you've seen my comment above :)

**Comment by andrewebdev** [comment]
Hey @notwaldorf I did indeed see your comment; It just so happened that it all happened as I started holiday :)
I'll revisit this in 1.5 week's time :)

**Comment by notwaldorf** [comment]
@andrewebdev Ah sure, no worries. Ping me when you're back and want me to look at it (my PR dashboard is a disaster)! 🎉

**Comment by andrewebdev** [comment]
@notwaldorf I just reviewed your request above to also add the ability to override content-type; But noticed that iron-form actually already have this ability [1].

Am I misunderstanding your request? :)

[1] https://github.com/PolymerElements/iron-form/blob/master/iron-form.html#L56-L59

**Comment by notwaldorf** [comment]
@andrewebdev I just wanted you to update the comment (from "Specify..." to "Note: setting a Content-Type header here will override the value specified by the contentType property of this element.", so basically the comment from here: https://github.com/PolymerElements/iron-ajax/blob/master/iron-ajax.html#L99). 

The code part of the PR was fine. :)

**Comment by andrewebdev** [comment]
Ah, misunderstanding from my part then :)
Right I've updated the comment as requested @notwaldorf

**Comment by notwaldorf** [comment]
Thanks so much! (I completely lost track of this PR, again)

**Pull request merged**